--- ruby-1.9.3-p374.orig/ext/tk/extconf.rb	2011-06-29 10:11:19.000000000 -0400
+++ ruby-1.9.3-p374/ext/tk/extconf.rb	2013-02-21 03:03:14.648655576 -0500
@@ -612,7 +612,8 @@
           $INCFLAGS << " -I" << File.join(File.dirname(File.dirname(file)),"include") if is_win32?
         else
           tcllibs = append_library($libs, libname)
-          tcllibs = "-L#{tcldir.quote} -Wl,-R#{tcldir.quote} " + tcllibs
+          # Don't use rpath
+          # tcllibs = "-L#{tcldir.quote} -Wl,-R#{tcldir.quote} " + tcllibs
 
           # FIX ME: avoid pathname trouble (fail to find) on MinGW.
           $INCFLAGS << " -I" << File.join(File.dirname(tcldir),"include") if is_win32?
@@ -654,7 +655,8 @@
         else
           tklibs = append_library("", libname)
           #tklibs = append_library("", $1)
-          tklibs = "-L#{tkdir.quote} -Wl,-R#{tkdir.quote} " + tklibs
+          # Don't use rpath
+          # tklibs = "-L#{tkdir.quote} -Wl,-R#{tkdir.quote} " + tklibs
 
           # FIX ME: avoid pathname trouble (fail to find) on MinGW.
           $INCFLAGS << " -I" << File.join(File.dirname(tcldir),"include") if is_win32?
