--- /tmp/b/ruby.spec	2013-04-25 17:01:32.104683834 +0900
+++ /tmp/a/ruby.spec	2013-04-25 17:41:13.604685903 +0900
@@ -9,7 +9,9 @@
 %global ruby_version_patch_level %{major_minor_version}.%{teeny_version}.%{patch_level}
 %global ruby_abi %{major_minor_version}.1
 
-%global ruby_archive %{name}-%{ruby_version}-p%{patch_level}
+%global ruby_name ruby
+
+%global ruby_archive %{ruby_name}-%{ruby_version}-p%{patch_level}
 
 %global ruby_libdir %{_datadir}/%{name}
 %global ruby_libarchdir %{_libdir}/%{name}
@@ -29,7 +31,7 @@
 
 # The RubyGems library has to stay out of Ruby directory three, since the
 # RubyGems should be share by all Ruby implementations.
-%global rubygems_dir %{_datadir}/rubygems
+%global rubygems_dir %{_datadir}/rubygems1.9
 
 # Specify custom RubyGems root.
 %global gem_dir %{_datadir}/gems
@@ -49,7 +51,7 @@
 %global	_normalized_cpu	%(echo %{_target_cpu} | sed 's/^ppc/powerpc/;s/i.86/i386/;s/sparcv./sparc/')
 
 Summary: An interpreter of object-oriented scripting language
-Name: ruby
+Name: ruby19
 Version: %{ruby_version_patch_level}
 # Note:
 # As seen on perl srpm, as this (ruby) srpm contains several sub-components,
@@ -160,7 +162,7 @@
 This package includes the libruby, necessary to run Ruby.
 
 # TODO: Rename or not rename to ruby-rubygems?
-%package -n rubygems
+%package rubygems
 Summary:    The Ruby standard for packaging ruby libraries
 Version:    %{rubygems_version}
 Group:      Development/Libraries
@@ -172,12 +174,12 @@
 Provides:   ruby(rubygems) = %{version}-%{release}
 BuildArch:  noarch
 
-%description -n rubygems
+%description  rubygems
 RubyGems is the Ruby standard for publishing and managing third party
 libraries.
 
 
-%package -n rubygems-devel
+%package  rubygems-devel
 Summary:    Macros and development tools for packaging RubyGems
 Version:    %{rubygems_version}
 Group:      Development/Libraries
@@ -185,11 +187,11 @@
 Requires:   ruby(rubygems) = %{version}-%{release}
 BuildArch:  noarch
 
-%description -n rubygems-devel
+%description  rubygems-devel
 Macros and development tools for packaging RubyGems.
 
 
-%package -n rubygem-rake
+%package  rubygem-rake
 Summary:    Ruby based make-like utility
 Version:    %{rake_version}
 Group:      Development/Libraries
@@ -200,7 +202,7 @@
 Provides:   rubygem(rake) = %{version}-%{release}
 BuildArch:  noarch
 
-%description -n rubygem-rake
+%description  rubygem-rake
 Rake is a Make-like program implemented in Ruby. Tasks and dependencies are
 specified in standard Ruby syntax.
 
@@ -219,7 +221,7 @@
 from the terminal.
 
 
-%package -n rubygem-rdoc
+%package  rubygem-rdoc
 Summary:    A tool to generate HTML and command-line documentation for Ruby projects
 Version:    %{rdoc_version}
 Group:      Development/Libraries
@@ -234,7 +236,7 @@
 Obsoletes:  ruby-ri < %{version}
 BuildArch:  noarch
 
-%description -n rubygem-rdoc
+%description  rubygem-rdoc
 RDoc produces HTML and command-line documentation for Ruby projects.  RDoc
 includes the 'rdoc' and 'ri' tools for generating and displaying online
 documentation.
@@ -243,7 +245,7 @@
 %package doc
 Summary:    Documentation for %{name}
 Group:      Documentation
-Requires:   %{_bindir}/ri
+Requires:   %{_bindir}/ri1.9
 # TODO: It seems that ri documentation differs from platform to platform due to
 # some encoding bugs, therefore the documentation should be split out of this gem
 # or kept platform specific.
@@ -254,7 +256,7 @@
 This package contains documentation for %{name}.
 
 
-%package -n rubygem-bigdecimal
+%package  rubygem-bigdecimal
 Summary:    BigDecimal provides arbitrary-precision floating point decimal arithmetic
 Version:    %{bigdecimal_version}
 Group:      Development/Libraries
@@ -263,7 +265,7 @@
 Requires:   ruby(rubygems) >= %{rubygems_version}
 Provides:   rubygem(bigdecimal) = %{version}-%{release}
 
-%description -n rubygem-bigdecimal
+%description  rubygem-bigdecimal
 Ruby provides built-in support for arbitrary precision integer arithmetic.
 For example:
 
@@ -276,7 +278,7 @@
 conversion between base 10 and base 2.
 
 
-%package -n rubygem-io-console
+%package  rubygem-io-console
 Summary:    IO/Console is a simple console utilizing library
 Version:    %{io_console_version}
 Group:      Development/Libraries
@@ -284,12 +286,12 @@
 Requires:   ruby(rubygems) >= %{rubygems_version}
 Provides:   rubygem(io-console) = %{version}-%{release}
 
-%description -n rubygem-io-console
+%description  rubygem-io-console
 IO/Console provides very simple and portable access to console. It doesn't
 provide higher layer features, such like curses and readline.
 
 
-%package -n rubygem-json
+%package  rubygem-json
 Summary:    This is a JSON implementation as a Ruby extension in C
 Version:    %{json_version}
 Group:      Development/Libraries
@@ -298,14 +300,14 @@
 Requires:   ruby(rubygems) >= %{rubygems_version}
 Provides:   rubygem(json) = %{version}-%{release}
 
-%description -n rubygem-json
+%description  rubygem-json
 This is a implementation of the JSON specification according to RFC 4627.
 You can think of it as a low fat alternative to XML, if you want to store
 data to disk or transmit it over a network rather than use a verbose
 markup language.
 
 
-%package -n rubygem-minitest
+%package  rubygem-minitest
 Summary:    Minitest provides a complete suite of testing facilities.
 Version:    %{minitest_version}
 Group:      Development/Libraries
@@ -315,7 +317,7 @@
 Provides:   rubygem(minitest) = %{version}-%{release}
 BuildArch:  noarch
 
-%description -n rubygem-minitest
+%description  rubygem-minitest
 minitest/unit is a small and incredibly fast unit testing framework.
 
 minitest/spec is a functionally complete spec engine.
@@ -372,7 +374,8 @@
         --with-ruby_pc='%{name}.pc' \
         --disable-rpath \
         --enable-shared \
-        --disable-versioned-paths
+        --disable-versioned-paths \
+        --program-suffix=1.9
 
 # Q= makes the build output more verbose and allows to check Fedora
 # compiler options.
@@ -513,9 +516,9 @@
 %lang(ja) %doc COPYING.ja
 %doc GPL
 %doc LEGAL
-%{_bindir}/erb
-%{_bindir}/ruby
-%{_bindir}/testrb
+%{_bindir}/erb*
+%{_bindir}/ruby*
+%{_bindir}/testrb*
 %{_mandir}/man1/erb*
 %{_mandir}/man1/ruby*
 
@@ -535,9 +538,9 @@
 %{_libdir}/libruby.so
 # TODO
 # ruby.pc still needs fixing, see bug 789532 comment 8
-%{_libdir}/pkgconfig/ruby.pc
+%{_libdir}/pkgconfig/%{name}.pc
 %if 0%{?fedora} <= 18
-%{_libdir}/pkgconfig/ruby-%{major_minor_version}.pc
+%{_libdir}/pkgconfig/%{name}-%{major_minor_version}.pc
 %endif
 
 %files libs
@@ -698,8 +701,8 @@
 %exclude %{ruby_libarchdir}/tkutil.so
 %{ruby_libarchdir}/zlib.so
 
-%files -n rubygems
-%{_bindir}/gem
+%files  rubygems
+%{_bindir}/gem*
 %{rubygems_dir}
 %{gem_dir}
 %exclude %{gem_dir}/gems/*
@@ -716,24 +719,24 @@
 %exclude %{gem_dir}/specifications/rake-%{rake_version}.gemspec
 %exclude %{gem_dir}/specifications/rdoc-%{rdoc_version}.gemspec
 
-%files -n rubygems-devel
+%files  rubygems-devel
 %config(noreplace) %{_sysconfdir}/rpm/macros.rubygems
 
-%files -n rubygem-rake
-%{_bindir}/rake
+%files  rubygem-rake
+%{_bindir}/rake*
 %{gem_dir}/gems/rake-%{rake_version}
 %{gem_dir}/specifications/rake-%{rake_version}.gemspec
-%{_mandir}/man1/rake.1*
+%{_mandir}/man1/rake*
 
 %files irb
-%{_bindir}/irb
+%{_bindir}/irb*
 %{ruby_libdir}/irb.rb
 %{ruby_libdir}/irb
-%{_mandir}/man1/irb.1*
+%{_mandir}/man1/irb*
 
-%files -n rubygem-rdoc
-%{_bindir}/rdoc
-%{_bindir}/ri
+%files  rubygem-rdoc
+%{_bindir}/rdoc*
+%{_bindir}/ri*
 %{gem_dir}/gems/rdoc-%{rdoc_version}
 %{gem_dir}/specifications/rdoc-%{rdoc_version}.gemspec
 %{_mandir}/man1/ri*
@@ -745,26 +748,26 @@
 %doc doc/ChangeLog-*
 %{_datadir}/ri
 
-%files -n rubygem-bigdecimal
+%files  rubygem-bigdecimal
 %{ruby_libdir}/bigdecimal
 %{ruby_libarchdir}/bigdecimal.so
 %{_libdir}/gems/exts/bigdecimal-%{bigdecimal_version}
 %{gem_dir}/gems/bigdecimal-%{bigdecimal_version}
 %{gem_dir}/specifications/bigdecimal-%{bigdecimal_version}.gemspec
 
-%files -n rubygem-io-console
+%files  rubygem-io-console
 %{ruby_libdir}/io
 %{ruby_libarchdir}/io/console.so
 %{_libdir}/gems/exts/io-console-%{io_console_version}
 %{gem_dir}/gems/io-console-%{io_console_version}
 %{gem_dir}/specifications/io-console-%{io_console_version}.gemspec
 
-%files -n rubygem-json
+%files  rubygem-json
 %{_libdir}/gems/exts/json-%{json_version}
 %{gem_dir}/gems/json-%{json_version}
 %{gem_dir}/specifications/json-%{json_version}.gemspec
 
-%files -n rubygem-minitest
+%files  rubygem-minitest
 %{gem_dir}/gems/minitest-%{minitest_version}
 %{gem_dir}/specifications/minitest-%{minitest_version}.gemspec
 
